modules/src/clients/ics07_tendermint/client_def.rs

```diff
@@ -193,116 +197,198 @@ impl ClientDef for TendermintClient {
 
     fn verify_client_consensus_state(
         &self,
-        _client_state: &Self::ClientState,
-        _height: Height,
-        _prefix: &CommitmentPrefix,
-        _proof: &CommitmentProofBytes,
-        _client_id: &ClientId,
-        _consensus_height: Height,
-        _expected_consensus_state: &AnyConsensusState,
+        client_state: &Self::ClientState,
+        prefix: &CommitmentPrefix,
+        proof: &CommitmentProofBytes,
+        root: &CommitmentRoot,
+        client_id: &ClientId,
+        consensus_height: Height,
+        expected_consensus_state: &AnyConsensusState,
     ) -> Result<(), Ics02Error> {
-        todo!()
+        let path = Path::ClientConsensusState {
+            client_id: client_id.clone(),
+            epoch: consensus_height.revision_number,
+            height: consensus_height.revision_height,
+        }
+        .to_string();
+        let value = expected_consensus_state.encode_vec().unwrap();
```

Comment:
I'm not sure whether the coding should be hard-corded.

Reply:
Good question; I see that ibc-go uses a `codec.BinaryCodec` that gets passed into the verification functions. AFAIK, the spec assumes that details of encoding are communicated out-of-band, so it should be all right to hard-code (for now). But will let @adizere and @romac weigh in here. 🙏

Reply:
Actually, ICS24 explicitly states this -> 
> The provableStore:
> MUST use canonical data structure encodings provided in these specifications as proto3 files

Reply:
Good point! 👍 I have made a note of this in the meta issue #1696. Although in this particular case is, the `unwrap()` is alright because the `encode_vec()` method should actually be infallible (https://github.com/informalsystems/tendermint-rs/issues/1064).